src/java.base/share/classes/sun/security/ssl/SSLTransport.java

```diff
@@ -122,16 +123,35 @@ static Plaintext decode(TransportContext context,
         } catch (AEADBadTagException bte) {
             throw context.fatal(Alert.BAD_RECORD_MAC, bte);
         } catch (BadPaddingException bpe) {
-            /*
-             * The basic SSLv3 record protection involves (optional)
-             * encryption for privacy, and an integrity check ensuring
-             * data origin authentication.  We do them both here, and
-             * throw a fatal alert if the integrity check fails.
-             */
-             Alert alert = (context.handshakeContext != null) ?
-                     Alert.HANDSHAKE_FAILURE :
-                     Alert.BAD_RECORD_MAC;
-            throw context.fatal(alert, bpe);
+            // Check for unexpected plaintext alert message during TLSv1.3 handshake, @bug 8331682
+            if (srcsLength == 1 && !context.isNegotiated && context.handshakeContext != null &&
+                    ProtocolVersion.TLS13.equals(context.handshakeContext.negotiatedProtocol)) {
+                ByteBuffer packet = srcs[srcsOffset].duplicate();
+                packet.position(0);
+                byte contentType = packet.get();                   // pos: 0
+                byte majorVersion = packet.get();                  // pos: 1
+                byte minorVersion = packet.get();                  // pos: 2
```

Comment:
A bit of a nit: You might consider using `Record.getInt8()` here for these 1-byte calls.  It might require you to do a casts to byte since those getInt8 returns int, but the good thing is that if there's an attempted read off the end of the buffer you get SSLException which is consistent with the exception type you get from the `Record.getInt16()` on line 134.  The `Buffer.get()` calls throw BufferUnderflowException which is a RuntimeException and might be handled differently or in a different part of the code further up the stack than SSLException would be.

Reply:
Great suggestion, I'll update the code, thanks!